app/controllers/users/dossiers_controller.rb

```diff
@@ -360,12 +360,12 @@ def update_dossier_and_compute_errors
 
       if champs_params[:dossier]
         @dossier.assign_attributes(champs_params[:dossier])
-        # FIXME in some cases a removed repetition bloc row is submitted.
-        # In this case it will be trated as a new records and action will fail.
+        # FIXME: in some cases a removed repetition bloc row is submitted.
+        # In this case it will be treated as a new record, and the action will fail.
         @dossier.champs.filter(&:repetition?).each do |champ|
           champ.champs = champ.champs.filter(&:persisted?)
         end
-        if @dossier.champs.any?(&:changed?)
+        if @dossier.champs.any?(&:changed_for_autosave?)
```

Comment:
Removing the inverse relationship breaks the workaround we added in #3907 to fix the parent dossier not being touched in some cases (the workaround was to add an inverse relationship, but we now have to remove it).

The new workaround is to watch for `changed_for_autosave?` on champs. Unlike `changed?`, `changed_for_autosave?` also detects changes to attachments. This allows us to touch both `last_champ_updated_at` and `updated_at` in a single pass.